src/auth/src/credentials/mds_credential.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::credentials::traits::dynamic::Credential;
+use crate::credentials::Result;
+use crate::errors::CredentialError;
+use crate::token::{Token, TokenProvider};
+use async_trait::async_trait;
+use http::header::{HeaderName, HeaderValue, AUTHORIZATION};
+use lazy_static::lazy_static;
+use reqwest::header::HeaderMap;
+use reqwest::Client;
+use std::collections::HashMap;
+use std::env;
+
+const METADATA_FLAVOR_VALUE: &str = "Google";
+const METADATA_FLAVOR: &str = "metadata-flavor";
+
+lazy_static! {
+    // Use lazy_static to initialize the metadata URLs.
+    static ref _METADATA_ROOT: String = format!(
```

Comment:
Why a learning `_` in this variable name?

Reply:
Initially I labelled it thinking of it as a private variable like in python, but I realized that is invalid in rust, so updated it.